### PR TITLE
Register pre commit hook

### PR DESCRIPTION
**Pull request by cphyc**

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-   id: pre-commit-all-the-way-down
+    name: pre-commit-all-the-way-down
```
Comment (neutrinoceros) [inline .pre-commit-hooks.yaml:2]: the name (and id) of the hook should be "python-doc" (or `python-rst`) to leave room for other similar hooks

**Review by neutrinoceros — CHANGES_REQUESTED** [review]

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    Programming Language :: Python :: 3.9
+authors = Corentin Cadiou <contact@cphyc.me>
+
+[options]
+py_modules = pre_commit_all_the_way_down
+python_requires = >=3.6.1
+
+[options.entry_points]
+console_scripts =
+    pre-commit-all-the-way-down=pre_commit_all_the_way_down:main
```
Comment (neutrinoceros) [inline setup.cfg:24]: obviously this is no doing anything at the moment, but do you intend to actually have a console script here and what for ?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    rev: '5.8.0'
+    hooks:
+    - id: isort
+      additional_dependencies: [toml]
+-   repo: https://github.com/PyCQA/flake8
+    rev: '3.9.0'
+    hooks:
+    - id: flake8
+      additional_dependencies: [
+        flake8-bugbear>=20.3.2, # GH PR 2851
```
Comment (neutrinoceros) [inline .pre-commit-config.yaml:34]: 👼🏻 
```suggestion
        flake8-bugbear>=20.3.2,
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+repos:
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
+    hooks:
+    - id: trailing-whitespace
+    - id: end-of-file-fixer
+    - id: no-commit-to-branch
+      args: [--branch, main]
```
Comment (neutrinoceros) [inline .pre-commit-config.yaml:13]: btw this line isn't necessary any more: "main" and "master" are both protected by default
```suggestion
```

**Review by neutrinoceros — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (cphyc) [reply setup.cfg:24]: I had the (maybe wrong) impression it was required, or at least would simplify, calling it as a hook. This way, the `entry` in the `.pre-commit-hooks.yaml` file can simply be `pre-commit-all-the-way-down`

Comment (neutrinoceros) [reply setup.cfg:24]: well it's not *required* but it is certainly the easiest way to do it with a hook that's written in Python.
But my remark from the other PR also applies here: I don't think any actual program/hook should bear the name "pre-commit-all-the-way-down". If the one you're implementing in #1 is named "python-doc", it should also be the name of the entry point (as well as filename).

Comment (cphyc) [reply setup.cfg:23]: Fair enough. Changing it to `python-doc`.

**Pull request merged**